src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,47 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_CAPACITY = 10;
+    private int size = 0;
+    private final Entry<K, V>[] kyeValues = new Entry[MAX_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < MAX_CAPACITY; i++) {
+            if ((kyeValues[i] == null && i == 0)
+                    || (i > 0 && kyeValues[i] == null)
+                    || (key == null && kyeValues[i].getKey() == null)
+                    || (key != null && kyeValues[i].getKey() != null
+                    && kyeValues[i].getKey().equals(key))) {
+                kyeValues[i] = new Entry<>(key, value);
+                return;
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
            if (kyeValues[i].getKey() == key
                    || (kyeValues[i].getKey() != null && kyeValues[i].getKey().equals(key))) {
                kyeValues[i].setValue(value);
                return;
            }
        }
        kyeValues[size] = new Entry<>(key, value);
        size++;
```

нету смысла пробегаться по всей длинне масива, только по существующим элеметам. Мы должны проверить, нету ли в массиве ключа key, если есть, то заменить значение, если нету, то добавить в конец новую пару